src/solve-model.jl

```diff
@@ -1,171 +1,53 @@
 export solve_model!, solve_model
 
 """
-    solution = solve_model!(energy_problem[, optimizer; parameters])
+    solve_model!(energy_problem[, optimizer; parameters, save_solution = true])
 
-Solve the internal model of an `energy_problem`. The solution obtained by calling
-[`solve_model`](@ref) is returned.
+Solve the internal model of an `energy_problem`. If `save_solution`, then the
+solution and dual variables are computed and saved by [`save_solution!`](@ref).
 """
 function solve_model!(
     energy_problem::EnergyProblem,
     optimizer = HiGHS.Optimizer;
     parameters = default_parameters(optimizer),
+    save_solution = true,
 )
     model = energy_problem.model
     if model === nothing
         error("Model is not created, run create_model(energy_problem) first.")
     end
 
-    energy_problem.solution =
-        solve_model!(model, energy_problem.variables, optimizer; parameters = parameters)
+    solve_model(model, optimizer; parameters = parameters)
     energy_problem.termination_status = JuMP.termination_status(model)
-    if energy_problem.solution === nothing
+    if energy_problem.termination_status != JuMP.OPTIMAL
```

Comment:
This option in JuMP is more generic, I propose using it instead of only the optimal condition:

```suggestion
    if !JuMP.is_solved_and_feasible(model)
```